src/app/app.component.ts

```diff
@@ -57,7 +68,7 @@ export class AppComponent implements OnInit {
     this.store.dispatch(removeTodo(id));
   }
 
-  setCurrentTodo(id) {
-    this.currentTodoId = id;
+  selectTodo(id) {
+    this.store.dispatch(selectTodo(id));
```

Comment:
Rather than try to set and maintain local component state and try circumvent the redux flow, we replace this with an action dispatch